### DISCUSSION TITLE
Transactional Azure commands

### DISCUSSION THREAD

**Discussion by stijnmoreels** [arcus-azure/arcus.scripting#152] [Ideas]
Several of our scripts uses a combination of several Azure commands to be run as one single 'transaction' or 'unit'. In several of these cases, things gets added/inserted but it's maybe useful to have also for some of these a transactional 'undo' so we don't end up with a corrupt or 'partly done' result after the execution of the command.

**Comment by stijnmoreels** [comment]
Any comments, @mbraekman ?

**Reply by mbraekman** [reply]
It would definitely be a good idea to roll back any changes which have been done, in case of an issue.
Currently trying to think of which scripts are doing multiple actions (besides collecting the context of a service) but not sure where we do have it -> unless it's the Enable/Disable Logic App, in which case we need to be very careful, as in some cases we don't want it to try and rollback since, for example on the first run of the pipeline, the LA's it's trying to disable won't exist yet, so no rollback is required.

So, would be a good idea to perform a roll back in those situations where it 'makes sense', but we can scroll through them to find out.
